packages/flutter_driver/lib/src/extension.dart

```diff
@@ -242,6 +264,14 @@ class FlutterDriverExtension {
       return null;
   }
 
+  Future<WaitForAbsentResult> _waitForAbsent(Command command) async {
+    final WaitForAbsent waitForAbsentCommand = command;
+    if ((await _waitForAbsentElement(_createFinder(waitForAbsentCommand.finder))).evaluate().isNotEmpty)
```

Comment:
Actually, I think you don't need the `if` block at all. `_waitForAbsentElement` returns iff `Finder.precache` returns `false`, which means it found nothing. Otherwise, we rely on the timeout logic. So this method can be simplified to:

```dart
final WaitForAbsent waitForAbsentCommand = command;
await _waitForAbsentElement(_createFinder(waitForAbsentCommand.finder));
return new WaitForAbsentResult();
```